tests/background/memory_leak_test.py

```diff
@@ -0,0 +1,278 @@
+"""Memory leak detection tests for bucket polling background functions.
+
+This module contains a comprehensive but fast memory leak test designed for CI.
+It tests CurrentPoller memory usage with S3 connection pooling and garbage
+collection.
+"""
+
+import asyncio
+import gc
+import time
+from contextlib import asynccontextmanager
+from typing import Any, AsyncGenerator
+from unittest.mock import AsyncMock, patch
+
+import pytest
+from lsst.ts.rubintv.background.currentpoller import CurrentPoller
+from lsst.ts.rubintv.models.models import Channel, Location
+from lsst.ts.rubintv.models.models_init import ModelsInitiator
+
+from ..mockdata import RubinDataMocker
+
+
+class CIMemoryTestMocker(RubinDataMocker):
+    """Lightweight data mocker optimized for fast CI memory testing."""
+
+    def __init__(
+        self,
+        locations: list[Location],
+        metadata_entries_per_camera: int = 20,  # Reduced for CI speed
+        **kwargs: Any,
+    ) -> None:
+        """Initialize with modest data amounts for fast CI testing.
+
+        Parameters
+        ----------
+        locations : list[Location]
+            List of locations to mock data for
+        metadata_entries_per_camera : int
+            Number of metadata entries per camera (default: 20)
+        """
+        # Enable metadata creation with CI-appropriate amounts
+        kwargs.setdefault("include_metadata", True)
+        kwargs.setdefault("metadata_entries_per_camera", metadata_entries_per_camera)
+        super().__init__(locations, **kwargs)
+
+    def mock_up_data(self) -> None:
+        """Generate modest amounts of mock data for CI testing."""
+        super().mock_up_data()
+
+        # Add extra sequences for basic stress testing
+        for location in self._locations:
+            loc_name = location.name
+            for camera in location.cameras:
+                # Store original channels
+                original_channels = camera.channels[:]
+
+                # Create one additional channel per camera for basic testing
+                expanded_channels = []
+                for channel in original_channels:
+                    new_channel = Channel(
+                        name=f"{channel.name}_ci_test",
+                        title=f"{channel.title} CI Test",
+                        colour=channel.colour,
+                        per_day=channel.per_day,
+                    )
+                    expanded_channels.append(new_channel)
+
+                # Temporarily replace camera channels
+                original_camera_channels = camera.channels
```

Comment:
AFAIU this assignment is by reference and not by value. So if you alter `orginal_camera_channels` aren't you also altering camera.channels (and vice versa)?